src/lib/model/ng-artifacts-factory.ts

```diff
@@ -1,38 +1,63 @@
+
+    if (packageNameWithoutScope === flatModuleFileName) {
+      this._makeEsmPackageNameVal = `${flatModuleFileName}.js`;
+    } else {
+      const pathFromRoot = buildDirectory.replace(pathOffsetFromSourceRoot, '');
+      const modulePath = path.join(pathFromRoot, 'esm2015', `${flatModuleFileName}.js`);
+      const pkgName = pathExistsSync(modulePath) ? packageNameWithoutScope : flatModuleFileName;
+      this._makeEsmPackageNameVal = `${pkgName}.js`;
+    }
+
+    return this._makeEsmPackageNameVal;
+  }
+
   private _makeUmdPackageName(ngPkg: NgPackageData): string {
     return ngPkg.packageNameWithoutScope.replace(SCOPE_NAME_SEPARATOR, '-') + '.umd.js';
   }
 
   private _unixPathJoin(...paths: string[]): string {
-    const joined: string = path.join(...paths);
+    const joined = path.join(...paths);
     return path.ensureUnixPath(joined);
   }
 
   public calculateArtifactPathsForBuild(ngPkg: NgPackageData): NgArtifacts {
-    const pathFromRoot: string = path.resolve(ngPkg.buildDirectory, ngPkg.pathOffsetFromSourceRoot);
+    const { buildDirectory, pathOffsetFromSourceRoot, flatModuleFileName } = ngPkg;
+    const pathFromRoot = path.resolve(buildDirectory, pathOffsetFromSourceRoot);
 
     return {
-      main: path.join(ngPkg.buildDirectory,'bundles', this._makeUmdPackageName(ngPkg)),
-      module: path.join(ngPkg.buildDirectory, ngPkg.fullPackageName + '.es5.js'),
-      es2015: path.join(ngPkg.buildDirectory, ngPkg.fullPackageName + '.js'),
-      typings: path.join(pathFromRoot, ngPkg.flatModuleFileName + '.d.ts'),
-      metadata: path.join(pathFromRoot, ngPkg.flatModuleFileName + '.metadata.json')
+      main: path.join(buildDirectory, 'bundles', this._makeUmdPackageName(ngPkg)),
+      module: path.join(buildDirectory, 'esm5', this._makeEsmPackageName(ngPkg)),
+      es2015: path.join(buildDirectory, 'esm2015', this._makeEsmPackageName(ngPkg)),
+      typings: path.join(pathFromRoot, `${flatModuleFileName}.d.ts`),
+      metadata: path.join(pathFromRoot, `${flatModuleFileName}.metadata.json`)
     }
   }
 
   public calculateArtifactPathsForPackageJson(ngPkg: NgPackageData): NgArtifacts {
-    const rootPathFromSelf: string = path.relative(ngPkg.sourcePath, ngPkg.rootSourcePath);
+    const { sourcePath, rootSourcePath, flatModuleFileName } = ngPkg;
+    const rootPathFromSelf = path.relative(sourcePath, rootSourcePath);
 
     return {
       main: this._unixPathJoin(rootPathFromSelf, 'bundles', this._makeUmdPackageName(ngPkg)),
-      module: this._unixPathJoin(rootPathFromSelf, `${ngPkg.fullPackageName}.es5.js`),
-      es2015: this._unixPathJoin(rootPathFromSelf, `${ngPkg.fullPackageName}.js`),
-      typings: path.ensureUnixPath(`${ngPkg.flatModuleFileName}.d.ts`),
-      metadata: path.ensureUnixPath(`${ngPkg.flatModuleFileName}.metadata.json`)
+      module: this._unixPathJoin(rootPathFromSelf, 'esm5', this._makeEsmPackageName(ngPkg)),
```

Comment:
We might want to turn `esm5` and `esm2015` into constants so we never have a typo in just one of the places where these folders are used.